### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by blizzz**
- typed properties
- add missing type hints and return type hints
- add missing return
- use match instead of switch
- use str_starts_with instead of strpos
- remove unused code and properties
- remove duplicated condition
- solve deprecations, avoid using private classes
- fix jsonSerialize signature

editing a config  is currently broken on server master+26 and requires https://github.com/nextcloud/server/pull/37520. Blocks a release, but not merging.

P.S.: Even not release blocking, as adding and deleting configs still works.

### CODE REVIEW

File: lib/Command/Add.php [hunk 1]
```diff
@@ -393,17 +375,10 @@ protected function askUnsignedInt(string $subject, string $label, InputInterface
 		$input->setOption($subject, $helper->ask($input, $output, $q));
 	}
 
-	protected function stringNormalizer($input) {
+	protected function stringNormalizer(?string $input): string {
 		return $input ? trim($input) : '';
 	}
```
Comment (come-nc) [inline lib/Command/Add.php:380]: Normalizes '0' into '' most likely, not sure if it is a big deal.

File: lib/Service/LdapQuerent.php [hunk 2]
```diff
@@ -109,10 +111,10 @@ public function find(string $search, int $limit = 0): \Generator {
 	}
 
 	/**
-	 * @throws \Exception
+	 * @throws Exception
 	 */
 	protected function getClient(): Ldap {
-		if ($this->ldap instanceof Ldap) {
+		if (isset($this->ldap) && $this->ldap instanceof Ldap) {
```
Comment (come-nc) [inline lib/Service/LdapQuerent.php:117]: Why this change? It is typed as `?Ldap`, the checks are redundant, the `instanceof` should be enough.

**Review by come-nc — COMMENTED** [review]
:broom:

File: lib/Service/LdapQuerent.php (hunk 2 above)
Comment (blizzz) [reply lib/Service/LdapQuerent.php:117]: Of course that's also possible, then it needs to be initialized as null though, otherwise there'll be a fatal error for it being uninitialized.

File: lib/Command/Add.php (hunk 1 above)
Comment (blizzz) [reply lib/Command/Add.php:380]: Why would it?

Comment (come-nc) [reply lib/Command/Add.php:380]: '0' is falsy: https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting
Something like this would fix it:
```suggestion
	protected function stringNormalizer(?string $input): string {
		return is_string($input) ? trim($input) : '';
	}
```
Or 
```suggestion
	protected function stringNormalizer(?string $input): string {
		return ($input !== null) ? trim($input) : '';
	}
```

File: lib/Service/LdapQuerent.php (hunk 2 above)
Comment (blizzz) [reply lib/Service/LdapQuerent.php:117]: It depends, I'd say. Not necessary when you set it in the constructor.

Comment (come-nc) [reply lib/Service/LdapQuerent.php]: Setting it in the constructor also counts as initializing, this is what I meant, once constructor is finished properties must respect their types.

File: lib/Command/Edit.php [hunk 3]
```diff
@@ -233,20 +228,20 @@ private function yesOrNoNormalizer(string $input): ?bool {
 		return null;
 	}
 
-	private function stringNormalizer($input) {
+	private function stringNormalizer(?string $input): string {
 		return $input ? trim($input) : '';
```
Comment (come-nc) [inline lib/Command/Edit.php:232]:
```suggestion
		return ($input !== null) ? trim($input) : '';
```

**Review by come-nc — APPROVED** [review]
:+1:
